onnx/onnx.in.proto

```diff
@@ -389,6 +390,15 @@ message ModelProto {
   //
   // If this field is empty, the training behavior of the model is undefined.
   repeated TrainingInfoProto training_info = 20;
+
+  // A list of function protos local to the model.
+  //
+  // The opset imports in FunctionProto should not conflict with the ones in 
+  // model proto and should be mergeable. Example, if the function proto imports domain  
```

Comment:
Nit: Can two different `FunctionProto`s in this list have conflicting imports? (I'm guessing they can't, but the wording here is ambiguous.)

Reply:
Also, is it useful to be able to control the opset imports on a per-function basis, given this restriction? Would it be more ergonomic for the functions to inherit the imports from the containing ModelProto?

Reply:
Yes, this originates from the use of FunctionProto defined outside of models (in defining standard ops as functions). We could potentially distinguish between the two usages and restrict some fields to be used only in FunctionProtos outside of models.

Reply:
ONNX does not allow importing different versions of the same domain so there cannot be conflicting imports... 

opset imports were added to FunctionProto to allow functions to easily import opsets other than what the model imports... the restriction is only in the version of the domain being unique within the model...